core/src/main/kotlin/team/backend/config/ReactiveQueryConfig.kt

```diff
@@ -12,99 +12,64 @@ import org.springframework.context.annotation.Configuration
-            setProperty(Settings.URL, (dataSource.unwrap(HikariDataSource::class.java) as HikariDataSource).jdbcUrl)
-            setProperty(Settings.USER, (dataSource.unwrap(HikariDataSource::class.java) as HikariDataSource).username)
-            setProperty(Settings.PASS, (dataSource.unwrap(HikariDataSource::class.java) as HikariDataSource).password)
-            setProperty(Settings.HBM2DDL_AUTO, "create")
-        }
-
-        override fun getProperties(): Properties = internalProps
-
-        override fun getPersistenceProviderClassName(): String = ReactivePersistenceProvider::class.qualifiedName!!
-    }
-
     @Bean
-    fun queryFactory(
-        sessionFactory: Mutiny.SessionFactory,
-        subqueryCreator: SubqueryCreator
-    ): SpringDataHibernateMutinyReactiveQueryFactory {
+    fun queryFactory(sessionFactory: Mutiny.SessionFactory, subQueryCreator: SubqueryCreator): SpringDataHibernateMutinyReactiveQueryFactory {
         return SpringDataHibernateMutinyReactiveQueryFactory(
             sessionFactory = sessionFactory,
-            subqueryCreator = subqueryCreator
+            subqueryCreator = subQueryCreator
         )
     }
+}
 
-    private fun jpaVendorAdapters(): JpaVendorAdapter = HibernateJpaVendorAdapter().apply {
-        setDatabasePlatform("org.hibernate.dialect.MySQLDialect")
-    }
+class ReactivePersistenceInfo(
+    persistenceUnitInfo: PersistenceUnitInfo,
+    jpaPropertyMap: Map<String, Any>,
+):
+    PersistenceUnitInfo by persistenceUnitInfo {
 
-    private fun jpaProperties(): Properties = Properties().apply {
-            setProperty("hibernate.vertx.pool.connect_timeout", "30000")
-            setProperty("hibernate.vertx.pool.class", "team.backend.config.MySQLConnectionPool")
-            setProperty("hibernate.vertx.pool.configuration_class", "team.backend.config.VertxMySQLDBConnectionPoolConfiguration")
-            setProperty("hibernate.dialect", "org.hibernate.dialect.MySQL8Dialect")
-            setProperty("hibernate.show_sql", "true")
-            setProperty("jakarta.persistence.jdbc.url", "jdbc:mysql://localhost:3306/chaterview")
-            setProperty("jakarta.persistence.jdbc.user", "doge")
-            setProperty("jakarta.persistence.jdbc.password", "1234")
-            setProperty("hibernate.use_sql_comments", "true")
-            setProperty("hibernate.show_sql", "true")
-            setProperty("hibernate.use_sql_comments", "true")
-            setProperty("hibernate.query.in_clause_parameter_padding", "true")
-            setProperty("hibernate.query.default_batch_fetch_size", "1000")
-    }
+    private val internalProps = Properties(persistenceUnitInfo.properties)
+        .apply {
+            putAll(jpaPropertyMap)
+            setProperty(Settings.SQL_CLIENT_POOL, MysqlConnectionPool::class.qualifiedName)
+            setProperty(
+                Settings.URL,
+                persistenceUnitInfo.nonJtaDataSource.unwrap(HikariDataSource::class.java).jdbcUrl
+            )
```

Comment:
I think it use `HikariCP`. Did you delete the DataSourceConfig class because of a setup problem ?

Reply:
Application started log.

<img width="911" alt="Screen Shot 2023-05-08 at 22 12 07" src="https://user-images.githubusercontent.com/47518272/236833289-d7d6fb8e-6dae-4fbe-983d-0a956fdd6aa7.png">

If you didn't delete it because of a setup problem, I think you should save DataSourceConfig.

Reply:
DataSourceConfig causes an error.
Even though I configured the yml file, the URL mapping still seems to be incorrect. Would you like to try?